File: Client/src/app/models/speaker.ts
import { Address } from "./address"
import { Career } from "./career"
import { School } from "./school"

export interface Speaker {
    id: number
    firstName: string
    middleName: string
    lastName: string
    title: string
    company: string
    schoolLastSpokeAt: School
    portraitUrl: string
    email: string
    phoneNumbers: PhoneNumber[]
    address: Address
    careers: Career[]
    careerIds: number[]
    publicId: string
}

export interface PhoneNumber {
    type: number
    number: string
    ext: string
    isPrimary: boolean
}

export interface SpeakerParams {
    searchTerm?: string
    pageNumber: number
    pageSize: number
}

File: CareerDayApi/Entities/Speaker.cs
namespace CareerDayApi.Entities
{
    public class Speaker
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }
        public string Title { get; set; }
        public string Company { get; set; }
        public string PortraitUrl { get; set; }
        public School SchoolLastSpokeAt { get; set; }
        public string Email { get; set; }
        public string PhoneNumber { get; set; }
        public SpeakerAddress Address { get; set; }
        public List<Career> Careers { get; set; }
        public string PublicId { get; set; }
    }
}